src/main/java/net/mcreator/ui/minecraft/ProcedureSelector.java

```diff
@@ -116,7 +117,7 @@ public ProcedureSelector(@Nullable IHelpContext helpContext, MCreator mcreator,
 		if (returnType == VariableElementType.LOGIC) {
 			defaultName = "(always)";
 			setBorder(BorderFactory
-					.createLineBorder(new Dependency("", VariableElementType.LOGIC.toDependencyType()).getColor()));
+					.createLineBorder(new Dependency("", VariableElementType.LOGIC.getName()).getColor()));

```

Comment:
Consider checking if returnType is not null in separate if and setting border there, so all dependency type colors are covered